exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -695,6 +698,33 @@ public void runQuery(QueryType type, List<PlanFragment> planFragments, UserResul
   }
 
   /**
+   * Get server properties that represent the list of server session options.
+   *
+   * @return server properties for the server session options.
+   */
+  public ServerProperties getOptions() throws RpcException {
```

Comment:
Sorry, it took me longer than a week but PR #764 contains API change for server metadata support with C++ client/JDBC driver support. If approved, it should make things way easier for you as the only change you would need is to update the server metadata to get the quoting information from the session.